src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/PartialSuccessTransmissionPolicy.cs

```diff
@@ -0,0 +1,153 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel.Implementation
+{
+    using System;
+    using System.Threading.Tasks;
+
+    using System.Web.Script.Serialization;
+
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+
+#if NET45
+    using TaskEx = System.Threading.Tasks.Task;
+#endif
+
+    internal class PartialSuccessTransmissionPolicy : TransmissionPolicy, IDisposable
+    {
+        private readonly JavaScriptSerializer serializer = new JavaScriptSerializer();
+
+        private TaskTimer pauseTimer = new TaskTimer { Delay = TimeSpan.FromSeconds(SlotDelayInSeconds) };
+
+        public override void Initialize(Transmitter transmitter)
+        {
+            base.Initialize(transmitter);
+            transmitter.TransmissionSent += this.HandleTransmissionSentEvent;
```

Comment:
Looks like every policy is subscribing for transmitter.TransmissionSent. Why not move it base class and make HandleTransmissionSentEvent method protected virtual?